scene/main/node.cpp

```diff
@@ -1715,8 +1715,41 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = data.owner->is_ancestor_of(p_parent);
```

Comment:
```suggestion
	bool preserve_owner = data.owner && data.owner->is_ancestor_of(p_parent);
```
Otherwise it will crash with no owner.

Reply:
ah, yes, my old friend the segfault. in retrospect it's obvious to check for, but i would be blind if it weren't for hindsight! i appreciate your eyes on this.

Reply:
tbh this one was caught by CI.